src/watershed.jl

```diff
@@ -39,22 +52,63 @@ function watershed(img::AbstractArray{T, N}, markers::AbstractArray{S,N}) where
             for j in CartesianIndices(_colon(max(Istart,i-one(i)), min(i+one(i),Iend)))
                 if segments[j] == 0
                     segments[j] = markers[i]
-                    enqueue!(pq, j, PixelKey(img[i], time_step))
+                    enqueue!(pq, j, PixelKey(compact ? Float64.(img[1]) : img[i], time_step, j))
```

Comment:
Is the `img[1]` a bug? Should it be `img[i]`? See also the `PK` suggestion above.